selfdrive/ui/qt/onroad.cc

```diff
@@ -404,11 +404,13 @@ void AnnotatedCameraWidget::drawText(QPainter &p, int x, int y, const QString &t
 }
 
 void AnnotatedCameraWidget::drawIcon(QPainter &p, int x, int y, QPixmap &img, QBrush bg, float opacity) {
+  p.save();
   p.setPen(Qt::NoPen);
   p.setBrush(bg);
   p.drawEllipse(x - radius / 2, y - radius / 2, radius, radius);
   p.setOpacity(opacity);
   p.drawPixmap(x - img_size / 2, y - img_size / 2, img);
+  p.restore();
```

Comment:
@adeebshihadeh setting opacity back to 100% will also solve this, not sure which is preferred?

Reply:
both solutions depend on the state of the painter that's passed in. instead, the opacity should be set before drawEllipse